src/utils/seedRefHooks.ts

```diff
@@ -0,0 +1,43 @@
+import { useState } from 'react';
+
+import { SeedRef } from 'utils/native';
+
+type IsValidFunc = () => boolean;
+type TryCreateFunc = (encryptedSeed: string, password: string) => Promise<void>;
+type TryDestroyFunc = () => Promise<void>;
+type TrySignFunc = (message: string) => Promise<string>;
+
+type SeedRefHooks = [IsValidFunc, TryCreateFunc, TryDestroyFunc, TrySignFunc];
+
+export function useSeedRef(): SeedRefHooks {
+	const [seedRef] = useState<SeedRef>(new SeedRef());
```

Comment:
if the `setSeedRef` is not defined here and it is not in the function closure, I think `seedRef` will not be updated when the object changed. That might be a problem.